gh-92240 : Include release dates for "What's New In Python 3.X"

georgically opened the pull request:
Added release dates for
"What's New in Python 3.X" for 3.0, 3.1, 3.2, 3.8 and 3.10

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->

#92240

georgically committed:
Include release dates for "What's New In Python 3.X"- (#92240)

Added release dates for
"What's New in Python 3.X" for 3.0, 3.1, 3.2, 3.8 and 3.10

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

slateny commented on Doc/whatsnew/3.0.rst:
```diff
@@ -53,9 +53,9 @@
-*intentionally backwards incompatible* Python release.  There are more
-changes than in a typical release, and more that are important for all
-Python users.  Nevertheless, after digesting the changes, you'll find
+*intentionally backwards incompatible* Python release. Python 3.0 was released on 3 Dec 2008.
```
I think that the date formatting should be (somewhat) consistent, at least for the whatsnew pages changed in this PR. Looking at [3.9's whatsnew](https://docs.python.org/3/whatsnew/3.9.html), it has it as 'October 5th, 2020', and 3.7 has it as 'June 27, 2018', so `month, day, year` seems like a good enough standard to go off in this case.

georgically replied:
I agree with date formatting consistency. Made adjustments to all versions.

georgically committed:
Edited date formatting for all what's new versions

2.1 - 'made' changed to 'released'
2.3, 2.6, 3.2  - whitespace and punctuation adjusted.
2.5 - release date added.
3.0, 3.1, 3.10, 3.8, 3.9 - date format adjusted.

georgically committed:
Merge branch '#92240-whats-new' of https://github.com/georgically/cpython into #92240-whats-new

georgically committed:
Merge branch 'main' into #92240-whats-new

AA-Turner commented on Doc/whatsnew/2.1.rst:
```diff
@@ -27,7 +27,7 @@ of new releases, with a new release coming every 6 to 9 months. 2.1 is the first
 January, 3 months after the final version of 2.0 was released.
 
-The final release of Python 2.1 was made on April 17, 2001.
+The final release of Python 2.1 was released on April 17, 2001.
```
"release ... was released" doesn't scan terribly well. I don't think this line needs to change.

A

georgically replied:
This change was made to retain consistency throughout the entire documentation among different versions. TBH, it is definitely not made on April 17, 2001 rather released.

AA-Turner replied:
Changing the text of a 21-year old whatsnew document purely for consistency seems iffy, I remain -1 on it.

A

JelleZijlstra replied:
Agree, this change is unnecessary

georgically committed:
Update 2022-05-18-23-58-26.gh-issue-92240.bHvYiz.rst

georgically committed:
Update 2022-05-18-23-58-26.gh-issue-92240.bHvYiz.rst

AA-Turner commented on Doc/whatsnew/2.3.rst:
```diff
@@ -8,7 +8,7 @@
 .. $Id: whatsnew23.tex 54631 2007-03-31 11:58:36Z georg.brandl $
 
-This article explains the new features in Python 2.3.  Python 2.3 was released
+This article explains the new features in Python 2.3. Python 2.3 was released
```
Similarly, double spacing was commonly used in documents in the past (and still is by some authors) -- see the remainder of What's New in Python 2.3 where `.  ` (full stop with two literal space characters) is used frequently. Please revert this change.

A

AA-Turner replied:
& also please revert where you've changed a double space to a single space in other documents (e.g. 3.0).

georgically replied:
Reverted mentioned double space. 3.0 release date was added later because it was missing. But it's adjusted to double spacing now.

JelleZijlstra commented on Doc/whatsnew/3.0.rst:
```diff
@@ -53,9 +53,9 @@
-*intentionally backwards incompatible* Python release.  There are more
-changes than in a typical release, and more that are important for all
-Python users.  Nevertheless, after digesting the changes, you'll find
+*intentionally backwards incompatible* Python release. Python 3.0 was released on Dec 3, 2008.
```
```suggestion
*intentionally backwards incompatible* Python release. Python 3.0 was released on December 3, 2008.
```

JelleZijlstra commented on Doc/whatsnew/3.10.rst:
```diff
@@ -46,8 +46,7 @@
 
-This article explains the new features in Python 3.10, compared to 3.9.
-
+This article explains the new features in Python 3.10, compared to 3.9. Python 3.10 was released on Oct 4, 2021.
```
```suggestion
This article explains the new features in Python 3.10, compared to 3.9.
Python 3.10 was released on October 4, 2021.
```

JelleZijlstra commented on Doc/whatsnew/3.2.rst:
```diff
@@ -48,7 +48,7 @@
    when researching a change.
 
-This article explains the new features in Python 3.2 as compared to 3.1.  It
+This article explains the new features in Python 3.2 as compared to 3.1. Python 3.2 was released on 20 Feb, 2011. It
```
```suggestion
This article explains the new features in Python 3.2 as compared to 3.1.
Python 3.2 was released on February 20, 2011. It
```

JelleZijlstra commented on Doc/whatsnew/3.8.rst:
```diff
@@ -44,7 +44,7 @@
 :Editor: Raymond Hettinger
 
-This article explains the new features in Python 3.8, compared to 3.7.
+This article explains the new features in Python 3.8, compared to 3.7. Python 3.8 was released on Oct 14, 2019.
```
```suggestion
This article explains the new features in Python 3.8, compared to 3.7.
Python 3.8 was released on October 14, 2019.
```

JelleZijlstra commented on Misc/NEWS.d/next/Documentation/2022-05-18-23-58-26.gh-issue-92240.bHvYiz.rst:
```diff
@@ -0,0 +1,6 @@
+2.1 - changed 'made' to 'released' to retain consistency.
+2.3, 2.6, 3.2 - whitespace and punctuation adjusted.
+2.5 - release date added.
+3.9 - date format adjusted.
```
```suggestion
```

This is unnecessary detail. Arguably the news entry isn't needed at all.

JelleZijlstra reviewed (commented):
Thanks for your contribution!

georgically committed:
Update Doc/whatsnew/3.0.rst

Co-authored-by: Jelle Zijlstra <jelle.zijlstra@gmail.com>

georgically committed:
Update Doc/whatsnew/3.10.rst

Co-authored-by: Jelle Zijlstra <jelle.zijlstra@gmail.com>

georgically committed:
Update Doc/whatsnew/3.2.rst

Co-authored-by: Jelle Zijlstra <jelle.zijlstra@gmail.com>

georgically committed:
Update Misc/NEWS.d/next/Documentation/2022-05-18-23-58-26.gh-issue-92240.bHvYiz.rst

Co-authored-by: Jelle Zijlstra <jelle.zijlstra@gmail.com>

georgically committed:
Update Doc/whatsnew/3.8.rst

Co-authored-by: Jelle Zijlstra <jelle.zijlstra@gmail.com>

georgically committed:
Merge branch 'python:main' into #92240-whats-new

georgically committed:
undo whitespaces changes from 2.X

AA-Turner commented on Doc/whatsnew/3.1.rst:
```diff
@@ -46,7 +46,7 @@
    when researching a change.
 
-This article explains the new features in Python 3.1, compared to 3.0.
+This article explains the new features in Python 3.1, compared to 3.0. Python 3.1 was released on June 27, 2009.
```
```suggestion
This article explains the new features in Python 3.1, compared to 3.0.
Python 3.1 was released on June 27, 2009.
```

JelleZijlstra committed:
Update Doc/whatsnew/3.1.rst

Co-authored-by: Adam Turner <9087854+AA-Turner@users.noreply.github.com>
